### PR TITLE
Project Feature Additions

### PR DESCRIPTION
**Pull request by michaelwilner**
* Order Updates map to FIX `OrderCancelReplaceRequest`
    * See test @ [https://github.com/bitfinexcom/bfxfixgw/commit/16d8e16eac25194115e920a34361cc616a9f101c#diff-4d05be89d986a85e95da593447717941R323](https://github.com/bitfinexcom/bfxfixgw/commit/16d8e16eac25194115e920a34361cc616a9f101c#diff-4d05be89d986a85e95da593447717941R323)
* Dynamic Time in Force maps to common FIX fields
    * Presence indicated by TimeInForce=GoodTillDate `59=6`
    * Actual time in field ExpireTime=<date> `126=<date>`
    * Example: `59=6|126=20060102-15:04:05.000"` see [https://github.com/bitfinexcom/bfxfixgw/commit/bb1e65f2e084d8cb36b0f57d64b28760ad3b5956#diff-4d05be89d986a85e95da593447717941R140](https://github.com/bitfinexcom/bfxfixgw/commit/bb1e65f2e084d8cb36b0f57d64b28760ad3b5956#diff-4d05be89d986a85e95da593447717941R140)

### CODE REVIEW

File: convert/fix.go [hunk 1]
```diff
@@ -128,15 +128,27 @@ func BookActionToFIX(action bitfinex.BookAction) enum.MDUpdateAction {
...
+func TimeInForceToFIX(ordtype bitfinex.OrderType, mtstif int64) (enum.TimeInForce, time.Time) {
+	tif, ok := MTSToTime(mtstif)
+	if ok {
+		return enum.TimeInForce_GOOD_TILL_DATE, tif
+	}
 	switch ordtype {
 	case bitfinex.OrderTypeFOK:
 		fallthrough
 	case bitfinex.OrderTypeExchangeFOK:
-		return enum.TimeInForce_FILL_OR_KILL
```
Comment (JacobPlaster) [inline convert/fix.go]: Please could you add field mappings to the README

**Comment by JacobPlaster** [comment]
Not really sure if the `.travis` file is actually running anything (might be wrong). It would be nice to add this to our travis pipeline just to run the linting and unit tests.

**Comment by JacobPlaster** [comment]
I thin the default reconnect timeout and attempts are a bit low, maybe we can up these to something like:
```
p := websocket.NewDefaultParameters()
// Enable orderbook checksum verification
p.ReconnectAttempts = 100
p.ReconnectInterval = time.Second * 60
c := websocket.NewWithParams(p)
```

The bitfinex API rate limit window is between 40-60 seconds so setting the reconnect interval should ensure that we dont get the rate limit error.

**Comment by michaelwilner** [comment]
> Not really sure if the `.travis` file is actually running anything (might be wrong). It would be nice to add this to our travis pipeline just to run the linting and unit tests.

It is

**Comment by michaelwilner** [comment]
> I thin the default reconnect timeout and attempts are a bit low, maybe we can up these to something like:
> 
> ```
> p := websocket.NewDefaultParameters()
> // Enable orderbook checksum verification
> p.ReconnectAttempts = 100
> p.ReconnectInterval = time.Second * 60
> c := websocket.NewWithParams(p)
> ```
> The bitfinex API rate limit window is between 40-60 seconds so setting the reconnect interval to 1min should ensure that we dont get the rate limit error.

Done! Added in [https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/1d93477147235a0ece2086b5bc71759b7b3da46b](https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/1d93477147235a0ece2086b5bc71759b7b3da46b)

**Comment by michaelwilner** [comment]
> Please could you chaneg the make file to use `golangci-lint run` for linting? That will make it more consistent with out other go projects.
> 
> I've just ran it on this PR and there is not too many linting errors to fix up, maybe 20

Resolved in [https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/99ca3c6b3e8e16de303c85812dda513fa6c4b1ff](https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/99ca3c6b3e8e16de303c85812dda513fa6c4b1ff) and [https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/33138b11a69a73fb6263083c1fd2bc2e2ac7438d](https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/33138b11a69a73fb6263083c1fd2bc2e2ac7438d)

File: convert/fix.go (hunk 1 above)
Comment (michaelwilner) [reply convert/fix.go]: Done [https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/2a96c9f88e8a63e54801f5c4745201b5f837f117](https://github.com/bitfinexcom/bfxfixgw/pull/19/commits/2a96c9f88e8a63e54801f5c4745201b5f837f117)

**Comment by JacobPlaster** [comment]
Thanks for the good work @michaelwilner

**Pull request merged**